src/lib/optimizer/strategy/index_scan_rule.cpp

```diff
@@ -0,0 +1,76 @@
+#include "index_scan_rule.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "all_parameter_variant.hpp"
+#include "constant_mappings.hpp"
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "optimizer/table_statistics.hpp"
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+// Kind of arbitrarily chosen, but the following paper suggests something similar
+// Access Path Selection in Main-Memory Optimized Data Systems: Should I Scan or Should I Probe?
+constexpr float INDEX_SCAN_SELECTIVITY_THRESHOLD = 0.01f;
+
+// From GroupKey paper
+constexpr float INDEX_SCAN_ROW_COUNT_THRESHOLD = 1000.0f;
```

Comment:
The constants could use a comment. Is this the table size under/over which I don't do an index scan? Input or output?